lib/exfmt/ast/to_algebra.ex

```diff
@@ -501,7 +529,13 @@ defmodule Exfmt.Ast.ToAlgebra do
 
       # Zero arity call
       %{args: []} ->
-        concat(name, "()")
+        case nested_spec_lhs_call?(ctx) || includes_spec_rhs?(ctx) do
```

Comment:
I did some investigating and unfortunately ```:spec_rhs``` can be arbitrarily deep. For example:

```elixir
@spec run(String.t) :: atom | atom | atom | atom | atom | String.t | :hello | :world
```

produces this stack:

```elixir
ctx: %Exfmt.Context{
  stack: [:|, :|, :|, :|, :|, :|, :spec_rhs, :module_attribute, :do, :call]
}
```

I'll dedicate some time this weekend to thinking about this more.